uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/leave-request-popup-controller.js

```diff
@@ -266,6 +270,15 @@ define([
       };
 
       /**
+       * Checks if popup is opened in view mode
+       *
+       * @return {Boolean}
+       */
+      vm.isView = function () {
```

Comment:
Instead of `isView()` we can create `isMode()` and send params
Example: `isMode('edit')` or `isMode('view')` or `isMode('create')`

with this function we can get rid of all `vm.mode === '<mode name>'` code

Same goes for `role`